### DISCUSSION TITLE
Is there a way to notify other clients (a.k.a responders) when a particular client disconnected from nats server?

### DISCUSSION THREAD

**Discussion by skaluva** [nats-io/nats-server#4795] [General]
We have a usecase, where we need to inform other clients whenever disconnection happens to perform certain activities. Currently i see connection listener but it just notifies that client but can't find a way to inform other clients who are subscribed to it. I found a thread with similar requirement but it was asked in 2017, not sure its implemented by this time or not.

https://github.com/nats-io/nats-server/issues/586

In this thread there is an alternate approach suggested like polling "connz" , but i would like to know if there is any better approach by now?

**Comment by ColinSullivan1** [comment]
You could subscribe on the disconnect system event:
`$SYS.ACCOUNT.<id>.DISCONNECT` (client disconnects)

https://docs.nats.io/running-a-nats-service/configuration/sys_accounts#available-events-and-services

Note that this requires System account credentials. You can import/export this subject to another account to avoid having to share system credentials.
